File: 119.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex < 0:return []
        res = [1]
        for _ in range(rowIndex):
            temp = [0] + res + [0]
            res = [temp[i]+temp[i+1] for i in range(len(temp)-1)]
        return res

File: 119.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> np;
        np.resize(rowIndex + 1);
        for (int i = 0; i < rowIndex + 1; i++){
            np[i].resize(i+1);
            np[i][0] = 1;
            for(int col = 1 ; col <= i; col++){
                np[i][col] = np[i-1][col-1] + (col < i ? np[i-1][col] : 0);
            }
        }
        return np[rowIndex];
    }
};
